docs/source/add_python_operator.rst

```diff
@@ -1,4 +1,128 @@
 Adding a new operator
 =====================
 
-There are a few steps to add a new operator on the python side.
+To add a new operator to SMAUG, we first need to implement the actual operator
+in C++.  Please see the `C++ tutorial <doxygen_html/index.html>` for details.
+Once that is done, all that is left is to expose a Python API which will add
+the operator to the model.
+
+Because at their core, Python operators simply add nodes to the graph, we can
+build arbitrarily complex operators in Python that invoke other Python
+operators. We will start simple, and then demonstrate more complex examples.
+
+Adding a simple Python operator
+-------------------------------
+
+A simple Python operator is one that adds only a single underlying SMAUG
+operator. For the purposes of this tutorial, we'll continue using the
+`MyCustomOperator` operator we built in the C++ tutorial, which implements an
+elementwise add. This operator is as simple as it can get: it takes just two
+input tensors and no additional parameters and produces a single output tensor.
+
+First, go to `smaug/python/ops <doxygen_html/node_8proto_source.html>`. All
```

Comment:
sphinx seems to have its own way to link to a source file. 

`:source: ../smaug/core/node.proto`

Then we don't need to include the protos in doxygen.

Reply:
This link pointed to the wrong place. But it wouldn't be possible for Sphinx to link to a source file that it doesn't index in the first place.

Reply:
Got you.